File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    ArrayList<TreeNode> inorder; 
    public TreeNode bstToGst(TreeNode root) {
      inorder=new ArrayList<>();
        inorder(root);
        int sum=0;
        for(int i=inorder.size()-1;i>=0;i--){
         sum+=inorder.get(i).val;
         TreeNode current=inorder.get(i);
         current.val=sum;
        }
        return root;
    }

    private void inorder(TreeNode root){
      if(root==null) return;
      inorder(root.left);
      inorder.add(root);
      inorder(root.right);
    }
}

File: 1114-binary-search-tree-to-greater-sum-tree/binary-search-tree-to-greater-sum-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
  vector<TreeNode*> inorderNodes;
public:
    TreeNode* bstToGst(TreeNode* root) {
        inorder(root);
        int sum = 0;
        for(int i = inorderNodes.size() - 1; i >= 0; --i) {
            sum += inorderNodes[i]->val;
            TreeNode* current = inorderNodes[i];
            current->val = sum;
        }
        return root;
    }

    private:
    void inorder(TreeNode* root) {
        if(root == nullptr) return;
        inorder(root->left);
        inorderNodes.push_back(root);
        inorder(root->right);
    }
};